agent/xds/jwt_authn.go

```diff
@@ -70,37 +79,69 @@ func makeJWTAuthFilter(pCE map[string]*structs.JWTProviderConfigEntry, intention
 		Providers: providers,
 		Rules:     rules,
 	}
-	return makeEnvoyHTTPFilter("envoy.filters.http.jwt_authn", cfg)
+	return makeEnvoyHTTPFilter(jwtEnvoyFilter, cfg)
 }
 
-func collectJWTRequirements(i *structs.Intention) []*structs.IntentionJWTProvider {
-	var jReqs []*structs.IntentionJWTProvider
+func collectJWTAuthnProviders(i *structs.Intention) []*JWTAuthnProvider {
+	var reqs []*JWTAuthnProvider
 
 	if i.JWT != nil {
-		jReqs = append(jReqs, i.JWT.Providers...)
+		for _, prov := range i.JWT.Providers {
+			reqs = append(reqs, &JWTAuthnProvider{Provider: prov, ComputedName: prov.Name})
+		}
 	}
 
-	jReqs = append(jReqs, getPermissionsProviders(i.Permissions)...)
+	reqs = append(reqs, getPermissionsProviders(i.Permissions)...)
 
-	return jReqs
+	return reqs
 }
 
-func getPermissionsProviders(p []*structs.IntentionPermission) []*structs.IntentionJWTProvider {
-	intentionProviders := []*structs.IntentionJWTProvider{}
+func getPermissionsProviders(p []*structs.IntentionPermission) []*JWTAuthnProvider {
+	var reqs []*JWTAuthnProvider
 	for _, perm := range p {
 		if perm.JWT == nil {
 			continue
 		}
-		intentionProviders = append(intentionProviders, perm.JWT.Providers...)
+		for _, prov := range perm.JWT.Providers {
+			reqs = append(reqs, &JWTAuthnProvider{Provider: prov, ComputedName: makeComputedProviderName(prov.Name, perm)})
+		}
+	}
+
+	return reqs
+}
+
+func makeComputedProviderName(name string, perm *structs.IntentionPermission) string {
```

Comment:
This approach needs to uniqueify the name based on all [potential HTTP match criteria](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-intentions#sources-permissions-http): path, methods, and headers.

It may be simpler to use an independent value that's unique enough. Like we could just pass the index of the IntentionPermission (`okta_1`, `okta_2`, `auth0_3`)